spec/tictactoe_spec.rb

```diff
@@ -7,7 +7,25 @@
   let (:test_ttt) { TicTacToe.new(game_io, player1, player2) }
 
   it "runs the game loop " do
-    allow(test_ttt.game_loop).to receive(:game_loop) { }
-    test_ttt.game_loop
+    expect(test_ttt.game_loop).to receive(:game_loop).once
+    expect(game_io).to receive(:puts_message)
+    expect(game_io).to receive(:get_input).and_return("N")
+    test_ttt.play_game
+  end
+
+  it "plays the game twice " do
+    expect(test_ttt.game_loop).to receive(:game_loop).twice
+    allow(test_ttt).to receive(:new_game)
+    expect(game_io).to receive(:puts_message).twice
+    expect(game_io).to receive(:get_input).and_return("Y", "N")
+    test_ttt.play_game
+  end
+
+  it "gameboad is blank after new_game " do
+    expect(test_ttt.game_loop).to receive(:game_loop).twice
+    expect(test_ttt).to receive(:play_again?).and_return(true, false)
```

Comment:
I think this may be a little unclear, so just to clarify: when you `expect(something).to receive(:some_method)`, RSpec is actually stubbing that method under the hood—that's how it manages to count how many times the method is called. For example:

```ruby
class Foo
  def method_1
    2.times { method_2 }
    true
  end

  private

  def method_2
    puts "hello world"
  end
end


describe Foo do
  let(:foo) { Foo.new }

  it "this will print 'hello world' twice to the console when running specs" do
    expect(foo.method_1).to be true
  end

  it "this will NOT print 'hello world' to the console because the method is stubbed by RSpec" do
    expect(foo).to receive(:method_2).twice
    foo.method_1
  end
end
```